src/index.ts

```diff
@@ -44,3 +45,62 @@ export function getServiceData(apiName: string, beaconName: string, chain: strin
     },
   };
 }
+
+interface PrivateKeySender {
+  privateKey: string;
+}
+
+interface MnemonicSender {
+  mnemonic: string;
+  derivationPath?: string;
+}
+
+type SenderAccount = PrivateKeySender | MnemonicSender;
+
+export function isPrivateKeySender(senderAccount: SenderAccount): senderAccount is PrivateKeySender {
+  return Object.prototype.hasOwnProperty.call(senderAccount, 'privateKey');
+}
+
+interface WhitelistBeaconReaderResult {
+  beaconId: string;
+  beaconReaderAddress: string;
+  expirationTimestamp: number;
+  indefiniteWhitelistStatus: boolean;
+}
+
+export async function whitelistBeaconReader(
+  beaconId: string,
+  beaconReaderAddress: string,
+  chain: string,
+  providerUrl: string,
+  senderAccount: SenderAccount
+): Promise<WhitelistBeaconReaderResult> {
+  const goBeaconWhitelister = goSync(() =>
+    require(`@api3dao/utility-contracts/SelfServeRrpBeaconServerWhitelister/deployments/${chain}/SelfServeRrpBeaconServerWhitelister.json`)
+  );
+  if (!isGoSuccess(goBeaconWhitelister)) throw new Error(`No whitelister contract found for chain: ${chain}`);
+  const beaconWhitelister = goBeaconWhitelister.data;
+
+  const provider = new ethers.providers.JsonRpcProvider(providerUrl);
+  const wallet = isPrivateKeySender(senderAccount)
+    ? new ethers.Wallet(senderAccount.privateKey)
+    : ethers.Wallet.fromMnemonic(senderAccount.mnemonic, senderAccount.derivationPath);
+  const connectedWallet = wallet.connect(provider);
+  const whitelisterContract = new ethers.Contract(beaconWhitelister.address, beaconWhitelister.abi).connect(
+    connectedWallet
+  );
```

Comment:
Here you can pass the signer or provider as a third argument instead: https://docs.ethers.io/v5/api/contract/contract/#Contract--creating